### ISSUE TITLE
Internet Explorer ≤ 9 hack doesn't work

### ISSUE THREAD

**Issue by atk** [4ae9b8/browserhacks#157]
http://browserhacks.com/#hack-e6b158b7e27bcb157326348efb5c0bb2 seems to target all browsers including all ie versions.

**Comment by KittyGiraudel** [comment]
It has been removed.

**Comment by j9t** [comment]
Hey guys—I had reported this hack [1] and would have loved to help investigate this. I assume you had good reason to remove it but I’ll run some more tests, too, to see what happened (and determine whether the original article has to be updated, too).

@jeffclayton, did you test exclusively with netrenderer.com?

[1] https://github.com/4ae9b8/browserhacks/issues/131

**Comment by jeffclayton** [comment]
The problem is that we have not been able to get it to work in anything but one situation (my test site above) -- the problem is that when not in that test setup (appears to be quirksmode only) the hack does not separate browsers. All browsers appear to have the same result. Browserhacks.com is in the standard html5 mode so if that is indeed what is doing it for everything, then the test cannot work on this site.

**Comment by jeffclayton** [comment]
I have many browsers and multiple machines of different types to test with, so no netrenderer was only one sample.

**Comment by jeffclayton** [comment]
If you have differing results I would of course love to know what you have found.

**Comment by j9t** [comment]
TL;DR: The hack still works †.

I set up a test page at http://hell.meiert.org/core/html/hack-character-escape.html, which I then ran against both NetRenderer and BrowserStack.

The result: Every browser _except for_ IE ≤6 applies the selector (because it’s valid).

https://www.browserstack.com/screenshots/e261b346e75cc40e1e4bee7837104021bdfc310c shows the final BrowserStack test, set up only to confirm the IE 6/7 boundary.

I suggest to include the hack again in the database with this new information, while I update the original post.

† It should have probably raised a red flag here that Microsoft would have fixed such a bug for _all_ versions of IE retroactively :)

**Comment by jeffclayton** [comment]
Did you try it in other browsers than the ie ones? in my testing it worked in many other browsers, it did not select just that set. That is what I am referring to.

**Comment by jeffclayton** [comment]
In other words, another example... :root was great once for determining modern browsers, but since all browsers use it now it really doesn't have anything that it does to differentiate anything now.

**Comment by j9t** [comment]
Here are some more (random) tests: https://www.browserstack.com/screenshots/62ac2673f58d45222ae029b232a4b7da9b91ff4d.

It seems pretty safe (to me) to say that the hack does still work, that is, it rules out IE ≤6 as indicated.

(My thinking here is simply that we had this in the hacks list, it got removed on the assumption it wouldn’t work, it does work, let’s put it back in. If you and the team decide that other rules apply now, please go for it—it’s your show. Other than that I hope this all helped.)

**Comment by jeffclayton** [comment]
The original point of the hack was listed as ie9 or less, not ie6 or less if you read the title, but that is fine - if it is ie6 or less then that is great -- the testing process has found the right target set -- just not what the original thought was for. It may be worth checking if you remove the html5 doctype if it shows different results. It has been a while but I am pretty sure it did.

**Comment by j9t** [comment]
I had tested the page with XHTML (delivered as text/html) and the results were the same—because they were I reverted the page and didn’t store any results. If the standard for Browserhacks is to set up different test pages with different doctypes, please let me know which scenarios you normally test and I’ll see whether I can help you and the team to build additional test pages.

**Comment by jeffclayton** [comment]
Tracing the full testing of a hack is one perfect reason why we need these comments. I probably did not test ie6 but that is fine since it was supposed to be an ie9- hack so testing ie10,9,8,7 which are more often used now would have been enough to show that it did not do the ie <=9 it was hoped for. 

Finding one that it does work with properly is key, so in answering your question, your assistance is very welcome.

**Comment by jeffclayton** [comment]
The idea really is to show all hacks on an easy to use page - ones that require a different doctype or other format wouldn't work properly -- on my own page (browserstrangeness.com) I had to create an outside stylesheet link set for a few that did not work the same way. It looks right but they are not single-sheet hacks like this site has.

**Comment by j9t** [comment]
Side note: I don’t think this was “supposed to” be an  IE 9 hack—it just got categorized as such here :) (Back when I identified the possibility to use it, it worked for all IE, back then going to IE 7.)

Can you document what modes/doctypes you need a test for, and in what OS/UA combinations at minimum? Is there another issue perhaps which can serve as a role model? (Do we want to keep this all here or work in a new request—do you mind filing one?)

**Comment by jeffclayton** [comment]
Due to the fact that most people will be using the base html5 (plain) doctype hacks that go here probably need to just be using that one. If a hack doesn't seem to work as planned, it may be just that - it may have wanted an older non-html5 doctype or html addition (xmlns=??) for example. It may well be worth looking at.

**Comment by j9t** [comment]
Thanks for trying to work this out here :)

I’m a bit confused now though. Shouldn’t the test page and the documented results show that the hack does work as advertised, with the only exception that it doesn’t target IE ≤9, but IE ≤6?

Just for convenience, what we have is:
- a test at http://hell.meiert.org/core/html/hack-character-escape.html
- documented supporting results with https://www.browserstack.com/screenshots/e261b346e75cc40e1e4bee7837104021bdfc310c and https://www.browserstack.com/screenshots/62ac2673f58d45222ae029b232a4b7da9b91ff4d
- undocumented supporting results with local and NetRenderer tests that I performed (saying the same thing, the syntax is not understood only in IE ≤6)

For my own investigation I’m happy right now (see the updated article)… I’ll be there to help where I can but I don’t mind leaving things now to you and team to take further :)

**Comment by jeffclayton** [comment]
What you are doing is great. We have been using this forum for this purpose among other things. I would love to hear anyone else's thoughts on it as to what we could do for documentation as this site really only handles that in these back-end conversations.

**Comment by j9t** [comment]
Okay, then I’ll be on stand-by for a moment. Let me know what else you need if what’s available now is not enough, and I’ll see where else I can help.

**Comment by j9t** [comment]
What is the status on this, can this be reconsidered?

It’s an ancient hack by now but reviewing the case it seems we could/should have reopened this issue with the new information collected in 2016.

**Comment by jeffclayton** [comment]
I didnt use netrendered, I used an actual browser- honestly I almost never
use it. I have vm's of winXP and win7,win8,and 10 that I test with other
than my usual machine.

I have seen failures with netrenderer which is why

it would be interesting to note that some tests people do with the html5
dictype and the older doctype including transitional

**Comment by jeffclayton** [comment]
Use something better like browserstack, which I also use

**Comment by jeffclayton** [comment]
I don't mind us looking into it again though.

**Comment by jeffclayton** [comment]
Going to the test page that was posted lists it as an "ie6 and below" hack, not an ie9 and below hack as it was originally listed -- that makes more sense as I mentioned back then. When I first came on with this group. I was let know people didn't worry about ones older than ie6. I would not have tested earlier browsers at all.

>>>
>>>
>>> On Thu, Dec 19, 2019 at 5:52 PM Jens Oliver Meiert <

**Comment by jeffclayton** [comment]
For more explanation - while it worked for the ones below and including
ie9, in my testing it did not exclude many others so barring a
misconfiguration in my test page (which is entirely possible), that's what
the problem was. Hacks need to target specifics, and without too much
configuration (if they require a lot of extra effort to make them work that
people cannot reproduce) they also must rule out ones that are not listed.
They still have merit often in different hacks-- as a part of another newer
one.

**Comment by jeffclayton** [comment]
If memory serves, it was the lack of targeted support/deviation in any modern browsers that made it slip away. Thank you for reposting and reminding me.

**Comment by j9t** [comment]
Cheers Jeff. What do you think of reopening the issue, just so that it’s clear that it’s being re-evaluated? Let me know if and when I can help in any way—the test at https://hell.meiert.org/core/html/hack-character-escape.html is still up.
